openquake/mbt/tests/tools/fault_modeler/test_fault_modeling_utils.py

```diff
@@ -247,24 +247,39 @@ def test_calc_mfd_from_fault_params_double_gr(self):
 
         # mfd_type should be set to 'DoubleTruncatedGR' by default;
         # this is part of the test.
+        # 20190801 - added M_ref
+        # mdf_rates modified, using M_ref = 0 as default value
 
         mfd, seis_rate = fmu.calc_mfd_from_fault_params(
                                                 self.fault_1,
-                                                # mfd_type='DoubleTruncatedGR',
+                                                #mfd_type='DoubleTruncatedGR',
                                                 param_map=self.param_map,
                                                 defaults=fmu.defaults)
 
-        mfd_rates = [(6.05, 0.008792708455724264),
-                     (6.1499999999999995, 0.0069842965860807266),
-                     (6.25, 0.005547823978012335),
-                     (6.35, 0.004406793227015579),
-                     (6.45, 0.003500440284810207),
-                     (6.55, 0.002780498552191055),
-                     (6.65, 0.002208628506615345),
-                     (6.75, 0.001754375982821495),
-                     (6.85, 0.0013935503774772729),
-                     (6.95, 0.0011069364113408821),
-                     (7.05, 0.000879270845572433)]
+        #mfd_rates_old = [(6.05, 0.008792708455724264),
+        #             (6.1499999999999995, 0.0069842965860807266),
+        #             (6.25, 0.005547823978012335),
+        #             (6.35, 0.004406793227015579),
+        #             (6.45, 0.003500440284810207),
+        #             (6.55, 0.002780498552191055),
+        #             (6.65, 0.002208628506615345),
+        #             (6.75, 0.001754375982821495),
+        #             (6.85, 0.0013935503774772729),
+        #             (6.95, 0.0011069364113408821),
+        #             (7.05, 0.000879270845572433)]
+        # computed using M_ref = 0.0 as default value
```

Comment:
A M_ref = 0 is not scaling. I would prefer to use a M_ref = M_max - Delta where delta could the in the order of 5.

Reply:
I agree, we must use a larger value of M_ref, but I don't in favour of use M_max - Delta.
If M_max = 6.0 and Delta = 5.0, M_ref = 1.0. IMO a value fixed by the user is better and this could be around M_ref = 4.0/5.0. If you agree I'll use M_ref = 4.0

Reply:
If we use a delta value, by just fixing one parameter you will obtain the same behaviour for a set of faults. Otherwise,  if you use a given value this will have to be defined on a case-by-case basis.

Reply:
The M_ref must not be defined case-by-case, as the delta value it must be a "global" value to be used for the entire database of faults or a set of them during the calculation.
I don't like the fact that M_ref depends on M_max.
Richard will make some tests, let's see what comes from them.

Reply:
But this is indeed the problem since the effect of the m_min that you define depends on the value of the m_max (as it can be seen the plot you posted). The impact of the m_min on the rates 'disappears' only when the m_min is much lower than the m_max (m_min << m_max). Therefore, the only way to guarantee an asymptotic behaviour (i.e. when m_min = - Infinity) is to use a m_min that is at lower than m_max of a certain amount of magnitude units (as you clearly show in your plot). Taking an extreme case, if you have a fault with m_max equal to 4.0 a m_min = 0 wouldn't give you an MFD close to the asymptotic one.